tests/dataset_runner.py

```diff
@@ -22,6 +24,7 @@ def __init__(self, deployment):
         self.submission_id = None
         self.submission_envelope = None
         self.upload_credentials = None
+        self.upload_area_uuid = None
```

Comment:
if we need to make it as an instance variable, how about moving the `self.upload_area_uuid = self.upload_credentials.split(':')[4]` to here?

Reply:
Two things:
1 - It is good practice in python to initialize all instance variables in init.  My IDE complains if I don't, and from StackOverflow "A tool such as Pylint will warn when you're setting instance variables outside __init__."
2. We don't have the necessary data yet, to set the upload_area_uuid in __init__.   Only after `_get_upload_area_credentials` returns something non-None can we do this.